packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -752,6 +778,59 @@ export default async function initSanity(
   function getOrganizationsWithAttachGrant(organizations: ProjectOrganization[]) {
     return pFilter(organizations, (org) => hasProjectAttachGrant(org.id), {concurrency: 3})
   }
+
+  async function createOrAppendEnvVars(filename: string) {
+    const variables = {
+      SANITY_PROJECT_ID: projectId,
+      SANITY_DATASET: datasetName,
+    }
+
+    try {
+      // use vercel's framework detector
+      const framework: Framework | null = await detectFrameworkRecord({
+        fs: new LocalFileSystemDetector(workDir),
+        frameworkList: frameworks as readonly Framework[],
+      })
+
+      if (framework && framework.slug !== 'sanity' && framework.envPrefix) {
+        print(
+          `\nDetected framework ${chalk.blue(framework?.name)}, using prefix '${
+            framework.envPrefix
+          }'`
+        )
+      }
+
+      await writeToEnv(filename, variables, {
+        framework,
+      })
+    } catch (err) {
+      print(err)
+      throw new Error('An error occurred while creating .env', {cause: err})
+    }
+  }
+
+  async function writeToEnv(
+    filename: string,
+    object: Record<string, string>,
+    options = {} as {
+      framework?: Framework | null
+    }
+  ) {
+    const prefix = options.framework?.slug === 'sanity' ? '' : options.framework?.envPrefix ?? ''
+    const envVariables = Object.keys(object)
+      .map((key) => `${prefix}${key}="${object[key]}"`)
+      .join('\n')
+
+    try {
+      // TO-DO: instead of appending if file exists, let's check whether variables are already set so we can just update them
```

Comment:
I worry that the "just append it" approach will lead to confusion and potential undesired surprises (like writing to the wrong dataset). The config the studio will end up using will now be determined by how the env vars are parsed (does it pick the first or last in case of duplicates?). Ideally, we should ask the user if they want to replace any existing variables, or, at the very least show a clear warning after appending in case of duplicates.

Reply:
+1, think this should make it into the current iteration

Reply:
Replacements were supposed to be done at a later stage, but it turns out they are more significant than first thought. I'll get to it now.